src/main/java/edu/ucsb/ucsbcslas/controllers/ReactController.java

```diff
@@ -5,7 +5,7 @@
 
 @Controller
 public class ReactController {
-  @RequestMapping(value = {"/", "/{x:[\\w\\-]+}", "/{x:^(?!api$).*$}/**/{y:[\\w\\-]+}"})
+  @RequestMapping(value = {"/", "/{x:^(?!swagger-ui$)[\\w\\-]+}", "/{x:^(?!api$).*$}/**/{y:[\\w\\-]+}"})
```

Comment:
does anybody know what this regex is doing

Reply:
Honestly... not completely!   

So, I do know *some* of what it is doing.  It is trying to say something like: "use this controller for the home page (`/`) and every URL *except* those that start with /api and those that start with /swagger-ui.

But there is also some deep black magic going on that I don't fully understand.  I tried simplifying these in ways that I *thought* should work, to make them less .. weird... and more readable.  I was unsuccessful.

Reply:
To clarify: the original lines of code here were written by @scottpchow23 (who got them from idk where), and then I hacked at them to get swagger to work.